### PR TITLE
fix errors from go vet. add unit test framework

### PR DESCRIPTION
**Pull request by yuvalif**
this is to fix issue #5 . but after running "go vet" other issues were found and fixed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+all: build
+
+build:
+	go fmt ./pkg/...
+	go vet ./pkg/...
+	go install -v ./pkg/...
+
+test:
+	ginkgo ./pkg/...
```
Comment (phoracek) [inline Makefile]: please add a newline

**Review by phoracek — CHANGES_REQUESTED** [review]
Just a nit. Thanks for the test and concurrency fixes.

**Comment by yuvalif** [comment]
just review the makefile, rest were added under the vendor directory due to "go dep"

**Comment by phoracek** [comment]
Sorry I forgot one thing, would you please add the test to travis build?

**Comment by phoracek** [comment]
Last travis build passed although the tests left error logs, would you please look into that?

```
E0705 10:25:41.004525    3123 plugin.go:105] world: Failed to setup a DPI gRPC server: listen unix /var/lib/kubelet/device-plugins/fake-lister_world: bind: no such file or directory
E0705 10:25:41.006293    3123 manager.go:214] Failed to start plugin's "world" server, atempt 1 ouf of 3 waiting 3000000000 before next try: listen unix /var/lib/kubelet/device-plugins/fake-lister_world: bind: no such file or directory
E0705 10:25:41.005234    3123 plugin.go:105] hello: Failed to setup a DPI gRPC server: listen unix /var/lib/kubelet/device-plugins/fake-lister_hello: bind: no such file or directory
E0705 10:25:41.006320    3123 manager.go:214] Failed to start plugin's "hello" server, atempt 1 ouf of 3 waiting 3000000000 before next try: listen unix /var/lib/kubelet/device-plugins/fake-lister_hello: bind: no such file or directory
```

https://travis-ci.org/kubevirt/device-plugin-manager/builds/400347588?utm_source=github_status&utm_medium=notification

**Comment by yuvalif** [comment]
the test itself is passing. i didn't mock the backend, hence the errors.

**Comment by phoracek** [comment]
LGTM

**Pull request merged**